[bug] PyTorch Benchmark running but not Datago

Hello,

I greatly appreciate the awesome work you are doing on Datago.

When testing the `python/benchmark_filesystem.py` script, the PyTorch benchmark runs fine but not the Datago one.

```
AssertionError: No image - benchmark did not run

│    59 │   # Save the last image as a test                                                                                                                                      │
│ ❱  60 │   assert img is not None, "No image - benchmark did not run"                                                                                                           │
│    61 │   img.save("benchmark_last_image.png")       
```
```
Benchmarking torch dataloader

Example printed batch:
[(<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D88C10>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F26984F5E90>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D2F790>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D8FA50>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D8FE50>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D98250>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D98610>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D98A10>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D98E10>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D592D0>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D59650>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D99950>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D51DD0>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D52150>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D5A4D0>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D8A890>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D9AB90>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D9AED0>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D53350>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D9B650>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D9BA10>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D53E90>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695CEC1D0>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695CEC5D0>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F26984ECA90>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D64E50>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695CED150>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695CED510>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F26984F59D0>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D2DD90>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D2E1D0>, 0), (<PIL.Image.Image image mode=RGB size=512x512 at 0x7F2695D8E5D0>, 0)]

Torch FPS 14.06
```
Folder structure is `nvme_data/test_folder/images`.

Datago is installed through pypi: https://pypi.org/project/datago/

Python version is 3.11.10.

OS is Ubuntu 22.04.

Thank you,

Enrico

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -41,4 +41,4 @@ jobs:
           DATAROOM_TEST_SOURCE: ${{ secrets.DATAROOM_TEST_SOURCE }}
           DATAROOM_API_URL: ${{ secrets.DATAROOM_API_URL }}
 
-        run: cd tests && go test -race -v .
+        run: cd tests && go test -v .  # -race once we fix the concurrent log accesses

--- a/.gitignore
+++ b/.gitignore
@@ -25,3 +25,8 @@ go.work.sum
 .env
 
 .vscode
+
+# local builds
+build
+
+__pycache__

--- a/generate_python_package.sh
+++ b/generate_python_package.sh
@@ -12,7 +12,7 @@ rm -rf $DESTINATION
 
 # Build the python package via the gopy toolchain
 cd pkg
-gopy pkg -author="Photoroom" -email="team@photoroom.com" -url="" -name="datago" -version="0.0.1".
+gopy pkg -author="Photoroom" -email="team@photoroom.com" -url="" -name="datago" -version="0.0.1" .
 mkdir -p $DESTINATION/datago
 mv datago/* $DESTINATION/datago/.
 mv setup.py $DESTINATION/.

--- a/pkg/generator_filesystem.go
+++ b/pkg/generator_filesystem.go
@@ -42,8 +42,14 @@ func newDatagoGeneratorFileSystem(config GeneratorFileSystemConfig) datagoGenera
 	for _, ext := range supported_img_extensions {
 		extensionsMap.Add(ext)
 	}
+
+	if config.Rank >= config.WorldSize {
+		panic("Rank should be less than World Size. Maybe you forgot to define both ?")
+	}
+
 	fmt.Println("File system root directory", config.RootPath)
 	fmt.Println("Supported image extensions", supported_img_extensions)
+	fmt.Println("Rank and World Size", config.Rank, config.WorldSize)
 
 	return datagoGeneratorFileSystem{config: config, extensions: extensionsMap}
 }

--- a/python/benchmark_filesystem.py
+++ b/python/benchmark_filesystem.py
@@ -24,6 +24,8 @@ def benchmark(
         "source_config": {
             "page_size": 512,
             "root_path": root_path,
+            "rank": 0,
+            "world_size": 1,
         },
         "image_config": {
             "crop_and_resize": crop_and_resize,

